packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -293,7 +295,7 @@ contract Exchange is IExchange, Initializable, Ownable, UsingRegistry {
    * @notice If conditions are met, updates the Uniswap bucket sizes to track
    * the price reported by the Oracle.
    */
-  function _updateBucketsIfNecessary() private {
+  function _updateBucketsIfNecessary() private nonReentrant {
```

Comment:
This worries me. `exchange` can call this function. Calling one `nonReentrant` function from another is not supported (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol#L27). 

We should add unit tests that test calling `exchange` in a situation where buckets should be updated.

Reply:
so... _should_ `exchange` be calling this function?

I'm not clear on what exactly this problem is and how to handle it. My rough sense of what you're saying here is that adding the `nonReentrant` modifier is the wrong thing. But, since the reentrancy bug is still a thing we care about preventing, we'd have to do something else?

Reply:
Yes. Exchange should still be calling this.

>But, since the reentrancy bug is still a thing we care about preventing, we'd have to do something else?

In most of these cases, though Slither reports a *potential* reentrancy problem, we can be fairly confident that nothing will happen. Reentrancy is a problem when the called contract is untrusted. In most (if not all) of our cases, we're calling other contracts in the Celo system.